public/app/features/plugins/admin/components/UpdateAllModal.tsx

```diff
@@ -215,24 +222,46 @@ export const UpdateAllModal = ({ isOpen, onDismiss, plugins }: Props) => {
         newSelectedPlugins.delete(error.id);
         return newSelectedPlugins;
       });
+
+      setInRequest('');
     }
   }, [error, errorMap, inProgress, selectedPlugins]);
 
+  // update plugins one by one (only used in cloud)
+  useEffect(() => {
+    if (startSyncUpdate && !inRequest) {
+      for (let plugin of plugins) {
```

Comment:
wouldn't this loop send all plugins to install in a single go anyway?

Reply:
we break the loop as soon as we get a selected plugin to install.